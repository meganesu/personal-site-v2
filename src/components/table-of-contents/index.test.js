import React from "react"
import { render, screen } from "@testing-library/react"

import TableOfContents from "."

const tableOfContents = {
  items: [
    {
      url: "#introduction",
      title: "Introduction",
    },
    {
      url: "#a-quick-crash-course-on-rss-feeds",
      title: "A Quick Crash Course on RSS Feeds",
      items: [
        {
          url: "#how-does-an-rss-feed-work",
          title: "How does an RSS feed work?",
        },
        {
          url: "#what-does-an-rss-feed-actually-look-like",
          title: "What does an RSS feed actually look like?",
        },
      ],
    },
    {
      url: "#prerequisites",
      title: "Prerequisites",
    },
    {
      url: "#the-starter-code",
      title: "The Starter Code",
    },
    {
      url: "#adding-an-rss-feed",
      title: "Adding an RSS Feed",
      items: [
        {
          url: "#prerequisite-confirm-that-you-have-sitemetadata",
          title: "Prerequisite: Confirm that you have siteMetadata",
        },
        {
          url: "#step-1-install-gatsby-plugin-feed",
          title: "Step 1: Install gatsby-plugin-feed",
        },
        {
          url: "#step-2-configure-the-plugin-in-your-gatsby-configjs-file",
          title: "Step 2: Configure the plugin in your gatsby-config.js file",
          items: [
            {
              url: "#configuring-optionsquery-and-feedquery",
              title: "Configuring options.query and feed.query",
            },
            {
              url: "#configuring-serialize",
              title: "Configuring serialize",
            },
            {
              url: "#the-full-plugin-configuration",
              title: "The Full Plugin Configuration",
            },
          ],
        },
        {
          url: "#step-3-build-and-serve-your-site-locally-to-test-the-rss-feed",
          title:
            "Step 3: Build and serve your site locally to test the RSS feed",
        },
        {
          url: "#step-4-optional-add-a-link-to-your-rss-feed-to-your-site",
          title: "Step 4: (Optional) Add a link to your RSS feed to your site",
        },
      ],
    },
    {
      url: "#troubleshooting-errors",
      title: "Troubleshooting Errors",
    },
    {
      url: "#takeaways",
      title: "Takeaways",
    },
    {
      url: "#wrap-it-up",
      title: "Wrap It Up",
    },
    {
      url: "#additional-resources",
      title: "Additional Resources",
    },
  ],
}

describe("TableOfContents component", () => {
  it("renders nested links in the correct DOM structure", () => {
    const { container } = render(
      <TableOfContents tableOfContents={tableOfContents} />
    )

    expect(container.querySelector("ul")).toMatchSnapshot()
  })

  it("renders the correct number of links", () => {
    render(<TableOfContents tableOfContents={tableOfContents} />)
    const links = screen.getAllByRole("link")

    expect(links.length).toBe(19)
  })

  it("renders a link with the correct url and title", () => {
    render(<TableOfContents tableOfContents={tableOfContents} />)
    const link = screen.getByRole("link", {
      name: "Step 4: (Optional) Add a link to your RSS feed to your site",
    })

    expect(link.href).toBe(
      "http://localhost/#step-4-optional-add-a-link-to-your-rss-feed-to-your-site"
    )
  })
})
