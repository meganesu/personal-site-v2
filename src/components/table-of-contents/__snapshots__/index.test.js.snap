// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableOfContents component renders nested links in the correct DOM structure 1`] = `
<ul>
  <li
    class="link"
  >
    <a
      href="/#introduction"
    >
      Introduction
    </a>
  </li>
  <li
    class="link"
  >
    <a
      href="/#a-quick-crash-course-on-rss-feeds"
    >
      A Quick Crash Course on RSS Feeds
    </a>
  </li>
  <ul>
    <li
      class="link"
    >
      <a
        href="/#how-does-an-rss-feed-work"
      >
        How does an RSS feed work?
      </a>
    </li>
    <li
      class="link"
    >
      <a
        href="/#what-does-an-rss-feed-actually-look-like"
      >
        What does an RSS feed actually look like?
      </a>
    </li>
  </ul>
  <li
    class="link"
  >
    <a
      href="/#prerequisites"
    >
      Prerequisites
    </a>
  </li>
  <li
    class="link"
  >
    <a
      href="/#the-starter-code"
    >
      The Starter Code
    </a>
  </li>
  <li
    class="link"
  >
    <a
      href="/#adding-an-rss-feed"
    >
      Adding an RSS Feed
    </a>
  </li>
  <ul>
    <li
      class="link"
    >
      <a
        href="/#prerequisite-confirm-that-you-have-sitemetadata"
      >
        Prerequisite: Confirm that you have siteMetadata
      </a>
    </li>
    <li
      class="link"
    >
      <a
        href="/#step-1-install-gatsby-plugin-feed"
      >
        Step 1: Install gatsby-plugin-feed
      </a>
    </li>
    <li
      class="link"
    >
      <a
        href="/#step-2-configure-the-plugin-in-your-gatsby-configjs-file"
      >
        Step 2: Configure the plugin in your gatsby-config.js file
      </a>
    </li>
    <ul>
      <li
        class="link"
      >
        <a
          href="/#configuring-optionsquery-and-feedquery"
        >
          Configuring options.query and feed.query
        </a>
      </li>
      <li
        class="link"
      >
        <a
          href="/#configuring-serialize"
        >
          Configuring serialize
        </a>
      </li>
      <li
        class="link"
      >
        <a
          href="/#the-full-plugin-configuration"
        >
          The Full Plugin Configuration
        </a>
      </li>
    </ul>
    <li
      class="link"
    >
      <a
        href="/#step-3-build-and-serve-your-site-locally-to-test-the-rss-feed"
      >
        Step 3: Build and serve your site locally to test the RSS feed
      </a>
    </li>
    <li
      class="link"
    >
      <a
        href="/#step-4-optional-add-a-link-to-your-rss-feed-to-your-site"
      >
        Step 4: (Optional) Add a link to your RSS feed to your site
      </a>
    </li>
  </ul>
  <li
    class="link"
  >
    <a
      href="/#troubleshooting-errors"
    >
      Troubleshooting Errors
    </a>
  </li>
  <li
    class="link"
  >
    <a
      href="/#takeaways"
    >
      Takeaways
    </a>
  </li>
  <li
    class="link"
  >
    <a
      href="/#wrap-it-up"
    >
      Wrap It Up
    </a>
  </li>
  <li
    class="link"
  >
    <a
      href="/#additional-resources"
    >
      Additional Resources
    </a>
  </li>
</ul>
`;
